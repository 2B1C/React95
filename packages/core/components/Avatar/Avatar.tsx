import * as React from 'react';
import styled, { css } from '@xstyled/styled-components';

export type AvatarProps = {
   src?: string,
   alt?: string,
   circle?: boolean
   size?: string | number
   children?: React.ReactNode
} & React.HTMLAttributes<HTMLDivElement>

const Image = styled.img` 
  max-width: 100%;
`

const StyledAvatar =  styled.div<AvatarProps>`
 ${({circle, size}) => css`
   display: flex;
   align-items: center;
   justify-content: center;
 
   background-color: material;
   border-top: 2px solid;
   border-top-color: borderDark;
   border-right: 2px solid;
   border-right-color: borderLightest;
   border-bottom: 2px solid;
   border-bottom-color: borderLightest;
   border-left: 2px solid;
   border-left-color: borderDark;
   border-radius: ${circle ? '50%' : 0};
   font-weight: bold;
   overflow: hidden;

   width: ${size ? size : 48}px;
   height: ${size ? size : 48}px;
 `}
`

const Avatar = React.forwardRef<HTMLDivElement, AvatarProps>(({src, alt, circle, children, ...otherProps}, ref) => (
    <StyledAvatar circle={circle}  ref={ref} {...otherProps}>
        {src ? <Image src={src} alt={alt} /> :  children}
    </StyledAvatar>
));

export default Avatar;
