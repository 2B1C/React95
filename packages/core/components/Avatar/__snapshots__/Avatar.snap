// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Snapshots should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #c3c7cb;
  border-top: 2px solid;
  border-top-color: #868a8e;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 2px solid;
  border-bottom-color: #ffffff;
  border-left: 2px solid;
  border-left-color: #868a8e;
  border-radius: 0;
  font-weight: bold;
  overflow: hidden;
  width: 48px;
  height: 48px;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`<Avatar /> Snapshots should match snapshot in default avatar with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #c3c7cb;
  border-top: 2px solid;
  border-top-color: #868a8e;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 2px solid;
  border-bottom-color: #ffffff;
  border-left: 2px solid;
  border-left-color: #868a8e;
  border-radius: 0;
  font-weight: bold;
  overflow: hidden;
  width: 48px;
  height: 48px;
}

<div>
  <div
    class="c0"
  >
    <p>
      GG
    </p>
  </div>
</div>
`;

exports[`<Avatar /> Snapshots should match snapshot in default avatar with children and circle 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #c3c7cb;
  border-top: 2px solid;
  border-top-color: #868a8e;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 2px solid;
  border-bottom-color: #ffffff;
  border-left: 2px solid;
  border-left-color: #868a8e;
  border-radius: 50%;
  font-weight: bold;
  overflow: hidden;
  width: 48px;
  height: 48px;
}

<div>
  <div
    class="c0"
  >
    <p>
      GG
    </p>
  </div>
</div>
`;

exports[`<Avatar /> Snapshots should match snapshot in default avatar with src and alt prop 1`] = `
.c1 {
  max-width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #c3c7cb;
  border-top: 2px solid;
  border-top-color: #868a8e;
  border-right: 2px solid;
  border-right-color: #ffffff;
  border-bottom: 2px solid;
  border-bottom-color: #ffffff;
  border-left: 2px solid;
  border-left-color: #868a8e;
  border-radius: 0;
  font-weight: bold;
  overflow: hidden;
  width: 48px;
  height: 48px;
}

<div>
  <div
    class="c0"
  >
    <img
      alt="React95 brand logo"
      class="c1"
      src="https://github.com/React95.png"
    />
  </div>
</div>
`;
